### PR TITLE
Remove more files with optimize_vendor

### PR DESCRIPTION
**Pull request by blackcoder87**
# Description
- Remove more files with optimize_vendor
- Updated ilchCore in the config.php of the gallery module. It requires a new file, which is part of Ilch 2.1.45.
- "replaceVendorDirectory"-Vorbereitung
- Run the tests on PHP 8.2 too.

Used parts of an earlier pull request by @hhunderter.
https://github.com/IlchCMS/Ilch-2.0/pull/607

### CODE REVIEW

File: build/optimize_vendor.php [hunk 1]
```diff
@@ -185,3 +213,23 @@ function quoteForPattern($fileOrDir)
     count($filesToDelete),
     number_format($savedSpace / 1024, 2, '.', ' ')
 );
+
+// Further delete some handpicked files
+$savedSpace = 0;
+$removedFiles = 0;
+$rootPath = realpath(__DIR__ . '/../');
+$otherFilesToDelete = ['composer.json', 'composer.lock', 'CONTRIBUTING.md', 'README.md'];
```
Comment (Mairu) [inline build/optimize_vendor.php:221]: If I read that correctly, you want to remove files from ilch itself?
You can do that by adding them to the `.gitattributes` file with the `export-ignore` flag

I would also not remove the composer.json as it is small and contains useful information in my eyes.

Comment (blackcoder87) [reply build/optimize_vendor.php:221]: I only would like to remove these files in the release for users as the composer files are not needed and probably of little interest for a user. On the other hand I want these files in the version for developers (the one we have linked in the readme.md).
https://github.com/IlchCMS/Ilch-2.0/blob/master/README.md

Wouldn't export-ignore also remove these files from the source files attached to a release?
https://github.com/IlchCMS/Ilch-2.0/releases/tag/v2.1.44

Then these files would be even more "useless" as it wouldn't be possible to run composer on it without restoring the composer files first.

I just briefly looked export-ignore up.. so correct me if I am wrong.

**Review by hhunderter — COMMENTED** [review]
Ich finde sieht gut aus

File: build/optimize_vendor.php (hunk 1 above)
Comment (Mairu) [reply build/optimize_vendor.php:221]: You are right, they would be missing then from default archive.

Still a simpler approach could be to maintain an additional file which will be merged into the gitattributes before the release archive is created. But I am also fine with keeping it the way you did.

We could also move the .gitattributes file (or at least the export-ignore part of it), to have a full developer archive.

**Pull request merged**